lib/model/rwfolder.go

```diff
@@ -847,18 +835,7 @@ func (f *sendReceiveFolder) deleteFile(file protocol.FileInfo) {
 		return
 	}
 
-	cur, ok := f.model.CurrentFolderFile(f.folderID, file.Name)
-	if ok && f.inConflict(cur.Version, file.Version) {
```

Comment:
As noted in `deleteFile`, conflict check needs to be moved there or happen here and the result passed on.